eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/clustering_writer/MergeAccessionClusteringWriterTest.java

```diff
@@ -0,0 +1,505 @@
+import static uk.ac.ebi.eva.accession.clustering.test.VariantAssertions.assertClusteredVariantAccessionEqual;
+import static uk.ac.ebi.eva.accession.clustering.test.VariantAssertions.assertReferenceSequenceAccessionEqual;
+
+/**
+ * This class handles some scenarios of ClusteringWriter where redundant RSs are discovered and merged.
+ *
+ * That includes all the scenarios that involve writing to the Operation collections, e.g.
+ * clusteredVariantOperationEntity to register an RS merge.
+ * Other test classes in this folder take care of other scenarios.
+ */
+@RunWith(SpringRunner.class)
+@EnableAutoConfiguration
+@ContextConfiguration(classes = {ClusteredVariantAccessioningConfiguration.class, BatchTestConfiguration.class})
+@TestPropertySource("classpath:clustering-pipeline-test.properties")
+public class MergeAccessionClusteringWriterTest {
+
+    private static final String TEST_DB = "test-db";
+
+    private static final String CLUSTERED_VARIANT_COLLECTION = "clusteredVariantEntity";
+
+    private static final String SUBMITTED_VARIANT_COLLECTION = "submittedVariantEntity";
+
+    public static final String DBSNP_CLUSTERED_VARIANT_COLLECTION = "dbsnpClusteredVariantEntity";
+
+    public static final String DBSNP_SUBMITTED_VARIANT_COLLECTION = "dbsnpSubmittedVariantEntity";
+
+    public static final long EVA_CLUSTERED_VARIANT_RANGE_START = 3000000000L;
+
+    public static final long EVA_SUBMITTED_VARIANT_RANGE_START = 5000000000L;
+
+    @Autowired
+    private InputParameters inputParameters;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private ClusteredVariantAccessioningService clusteredVariantAccessioningService;
+
+    @Autowired
+    private ContiguousIdBlockRepository contiguousIdBlockRepository;
+
+    //Required by nosql-unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    private ClusteringWriter clusteringWriter;
+
+    private Function<ISubmittedVariant, String> hashingFunction;
+
+    private Function<IClusteredVariant, String> clusteredHashingFunction;
+
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(
+            MongoDbConfigurationBuilder.mongoDb().databaseName(TEST_DB).build());
+
+    @Before
+    public void setUp() {
+        mongoTemplate.getDb().drop();
+        clusteringWriter = new ClusteringWriter(mongoTemplate, clusteredVariantAccessioningService,
+                                                EVA_SUBMITTED_VARIANT_RANGE_START, EVA_CLUSTERED_VARIANT_RANGE_START);
+        hashingFunction = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+        clusteredHashingFunction = new ClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
+    }
+
+    @After
+    public void tearDown() {
+        mongoTemplate.getDb().drop();
+    }
+
+    @Test
+    @DirtiesContext
+    public void merge_eva_clustered_accession() throws Exception {
```

Comment:
Any reason we are eschewing Camel casing in this test class?

Reply:
I tried to stick with camelCase but it was too hard to read. The alternative to underscores are more comments (that we won't read and won't update), so I think this is the least bad option.